docs/topics/tour/kotlin-tour-intermediate-null-safety.md

```diff
@@ -0,0 +1,658 @@
+* `!is` checks if the object **doesn't** have the type and returns a boolean value.
+
+You've already seen this in practice when you learned about sealed classes in the [Open and other special classes](kotlin-tour-intermediate-classes-part-3.md) chapter.
+
+For example:
+
+```kotlin
+fun printObjectType(obj: Any) {
+    when (obj) {
+        is String -> println("It's a String with length ${obj.length}")
+        is Int -> println("It's an Integer with value $obj")
+        !is Double -> println("It's NOT a Double")
+        else -> println("Unknown type")
+    }
+}
+
+fun main() {
+    val myString = "Hello, Kotlin!"
+    val myInt = 42
+    val myDouble = 3.14
+    val myList = listOf(1, 2, 3)
+
+    // The type is String
+    printObjectType(myString)
+    // It's a String with length 14
+
+    // The type is Int
+    printObjectType(myInt)
+    // It's an Integer with value 42
+
+    // The type is List, so it's NOT a Double.
+    printObjectType(myList)
+    // It's NOT a Double
+
+    // The type is Double, so the else branch is triggered.
+    printObjectType(myDouble)
+    // Unknown type
+}
+```
+{kotlin-runnable="true" id="kotlin-tour-null-safety-casts"}
+
+To explicitly _cast_ an object to any other type you can use the `as` operator. This includes casting from a nullable 
+type to its non-nullable counterpart. If the cast isn't possible, the program crashes **at runtime**. This is why it's 
+called the **unsafe** cast operator.
+
+```kotlin
+fun main() {
+//sampleStart
+    val a: String? = null
+    val b = a as String
+
+    // Triggers an error at runtime
+    print(b)
+//sampleEnd
+}
+```
+{kotlin-runnable="true" validate="false" id="kotlin-tour-null-safety-as-operator"}
+
+To explicitly cast an object to a non-nullable type, but return `null` instead of throwing an error on failure, use the `as?`
+operator. Since the `as?` operator doesn't trigger an error on failure, it is called the **safe** operator.
+
+```kotlin
+fun main() {
+//sampleStart
+    val a: String? = null
+    val b = a as? String
+
+    // Returns null value
+    print(b)
+    // null
+//sampleEnd
+}
+```
+{kotlin-runnable="true" id="kotlin-tour-null-safety-safe-operator"}
+
+The `as?` operator can be combined with the Elvis operator `?:`, to reduce several lines of code to one. For example,
+the following `calculateTotalStringLength()` function that calculates the total length of all strings provided in a mixed list,
+can be reduced from:
+
+```kotlin
+fun calculateTotalStringLength(items: List<Any>): Int {
+    var totalLength = 0
+
+    for (item in items) {
+        totalLength += if (item is String) {
+            item.length
+        } else {
+            0  // Add 0 for non-String items
+        }
+    }
+
+    return totalLength
+}
+```
+
+To:
+
+```kotlin
+fun calculateTotalStringLength(items: List<Any>): Int {
+    return items.sumOf { (it as? String)?.length ?: 0 }
+}
+```
+
+In the first version, the example:
```

Comment:
It might be even better to have these lists with explanations right next to the code
Plus, visually is would help to avoid to consequent lists

Reply:
I restructured it as you suggest :)